cluster/manager.go

```diff
@@ -0,0 +1,229 @@
+package cluster
+
+import (
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/memberlist"
+	"github.com/raintank/worldping-api/pkg/log"
+)
+
+type ClusterManager struct {
+	sync.RWMutex
+	Peers map[string]Node
+	node  Node
+	list  *memberlist.Memberlist
+}
+
+func (c *ClusterManager) SetList(list *memberlist.Memberlist) {
+	c.Lock()
+	c.list = list
+	c.Unlock()
+}
+
+func (c *ClusterManager) ThisNode() Node {
+	c.RLock()
+	defer c.RUnlock()
+	return c.node
+}
+
+func (c *ClusterManager) PeersList() []Node {
+	c.RLock()
+	list := make([]Node, len(c.Peers))
+	i := 0
+	for _, n := range c.Peers {
+		list[i] = n
+		i++
+	}
+	c.RUnlock()
```

Comment:
in general i think it's a good idea to use `defer` even if in this case it makes effectively no difference. It might make a difference if another person comes in to edit this code and adds a `return` somewhere in the middle.